stdlib/public/runtime/Heap.cpp

```diff
@@ -108,13 +97,13 @@ void *swift::swift_slowAllocTyped(size_t size, size_t alignMask,
                                   MallocTypeId typeId) {
 #if SWIFT_STDLIB_HAS_MALLOC_TYPE
   if (__builtin_available(macOS 9998, iOS 9998, tvOS 9998, watchOS 9998, *)) {
-    void *p;
+    void *p = nullptr;
     // This check also forces "default" alignment to use malloc_memalign().
     if (alignMask <= MALLOC_ALIGN_MASK) {
-      p = malloc_type_zone_malloc(DEFAULT_ZONE(), size, typeId);
+      p = malloc_type_malloc(size, typeId);
     } else {
       size_t alignment = computeAlignment(alignMask);
-      p = malloc_type_zone_memalign(DEFAULT_ZONE(), alignment, size, typeId);
+      malloc_type_memalign(&p, alignment, size, typeId);
```

Comment:
Nit: use `malloc_type_aligned_alloc` to avoid `memalign`'s awkward interface (out parameter).

Reply:
And fixed.